R-package/R/lgb.Dataset.R

```diff
@@ -533,22 +533,47 @@ Dataset <- R6::R6Class(
 
     # Update parameters
     update_params = function(params) {
-
-      # Parameter updating
-      if (!lgb.is.null.handle(private$handle)) {
-        lgb.call(
-          "LGBM_DatasetUpdateParam_R"
-          , ret = NULL
-          , private$handle
+      if (lgb.is.null.handle(private$handle)) {
+        private$params <- modifyList(private$params, params)
+      } else {
+        call_state <- .Call(
+          "LGBM_DatasetUpdateParamWarning_R"
+          , lgb.params2str(private$params)
           , lgb.params2str(params)
+          , call_state
```

Comment:
where is this `call_state` coming from? It's not being passed into `update_params()`